app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabs.kt

```diff
@@ -0,0 +1,339 @@
+        fontSize = 12.sp,
+        maxLines = 1,
+        modifier = Modifier
+            .padding(PaddingValues(16.dp, 7.dp))
+            .fillMaxWidth()
+    )
+}
+
+@Composable
+@Preview
+private fun SyncedTabsNoTabsItemPreview() {
+    FirefoxTheme {
+        Box(Modifier.background(FirefoxTheme.colors.layer2)) {
+            SyncedTabsNoTabsItem()
+        }
+    }
+}
+
+@Composable
+@Preview
+private fun SyncedTabsListPreview() {
+    FirefoxTheme {
+        Box(Modifier.background(FirefoxTheme.colors.layer2)) {
+            SyncedTabsList(
+                getFakeSyncedTabList(),
+                object : SyncedTabsView.Listener {
+                    override fun onRefresh() = Unit
+
+                    override fun onTabClicked(tab: mozilla.components.browser.storage.sync.Tab) {
+                        println("Tab clicked")
+                    }
+                }
+            )
+        }
+    }
+}
+
+/**
+ * Converts a list of [SyncedDeviceTabs] into a list of [SyncedTabsListItem].
+ */
+fun List<SyncedDeviceTabs>.toComposeList() = asSequence().flatMap { (device, tabs) ->
+    // transform to sticky headers data here https://github.com/mozilla-mobile/fenix/issues/19942
+    val deviceTabs = if (tabs.isEmpty()) {
+        sequenceOf(SyncedTabsListItem.NoTabs)
+    } else {
+        tabs.asSequence().map {
+            SyncedTabsListItem.Tab(it)
+        }
+    }
+
+    sequenceOf(SyncedTabsListItem.Device(device.displayName, device.id)) + deviceTabs
+}.toList()
+
+/**
+ * Converts the error type to the appropriate matching string resource for displaying to the user.
+ */
+fun SyncedTabsView.ErrorType.toStringRes() = when (this) {
+    SyncedTabsView.ErrorType.MULTIPLE_DEVICES_UNAVAILABLE -> R.string.synced_tabs_connect_another_device
+    SyncedTabsView.ErrorType.SYNC_ENGINE_UNAVAILABLE -> R.string.synced_tabs_enable_tab_syncing
+    SyncedTabsView.ErrorType.SYNC_UNAVAILABLE -> R.string.synced_tabs_sign_in_message
+    SyncedTabsView.ErrorType.SYNC_NEEDS_REAUTHENTICATION -> R.string.synced_tabs_reauth
+    SyncedTabsView.ErrorType.NO_TABS_AVAILABLE -> R.string.synced_tabs_no_tabs
+}
+
+/**
+ * Helper function to create a List of [SyncedTabsListItem] for previewing.
+ */
+@VisibleForTesting internal fun getFakeSyncedTabList(): List<SyncedTabsListItem> = listOf(
+    SyncedTabsListItem.Device("Device 1", ""),
+    SyncedTabsListItem.Tab(generateFakeTab("Mozilla", "www.mozilla.org")),
+    SyncedTabsListItem.Tab(generateFakeTab("Google", "www.google.com")),
+    SyncedTabsListItem.Device("Device 2", ""),
+    SyncedTabsListItem.NoTabs,
+    SyncedTabsListItem.Device("Device 3", ""),
+    SyncedTabsListItem.Error(SyncedTabsView.ErrorType.NO_TABS_AVAILABLE),
+)
+
+/**
+ * Helper function to create a [SyncedTabsListItem.Tab] for previewing.
+ */
+private fun generateFakeTab(tabName: String, tabUrl: String): SyncTab = SyncTab(
+    listOf(TabEntry(tabName, tabUrl, null)),
+    0,
+    0L
+)
```

Comment:
Could we put these in the test package itself? Does it need to be in our public packages (even though the have internal/private modifiers)?

Reply:
Can stuff in the test package be accessed here? Legitimately do not know haha

Reply:
I only noticed the references to the tests initially 🙂  - I don't have strong opinions about adding fake helpers to production code, but it _feels_ like code smell.